examples/convergence_study.jl

```diff
@@ -0,0 +1,82 @@
+# # Performing a convergence study
+#
+# This example shows how to perform a convergence study to find the 
+# appropriate kgrid and Ecut for a desired convergence and error tolerance.
+#
+# Generally, to wisely vary kgrid and Ecut, one can perform a convergence study by
+# starting with a reasonable grid or Ecut, picking a property such as energy, and increasing the grid
+# until the property converges – its value does not change much. 
+
+# This study must be performed separately for each value you wish to study (kgrid, Ecut, and even temperature).
+# In this example we will use a Pt system to perform a convergence study for kgrid and Ecut in DFTK.
+#
+# We start by setting up a simple cubic Pt system
+
+using DFTK
+using Plots
+using Unitful
+using UnitfulAtomic
+using LinearAlgebra
+using Optim
+
+kgrid = [1, 1, 1]       # k-point grid
+tol = 1e-2              # tolerance for the optimization routine
+Ecut = 490u"eV"
+temperature = 0.01
+Pt = ElementPsp(:Pt, psp=load_psp("hgh/lda/Pt-q10"));
+atoms = [Pt]
+position = [zeros(3)]
+a = 5.00 # in Bohr
+lattice = a * I(3)
+
+# Then we define a function to compute the total energy of the system for varying values of `Ecut` and `kgrid`
+function compute_Ecut_convergence(Ecut)
```

Comment:
Why not have one function to construct the basis (which takes Ecut, kgrid, tol etc as input parameters) and move all the global variables inside? That's a lot cleaner.